custom/icds_reports/management/commands/run_aggregation_query.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import absolute_import, print_function, unicode_literals
+
+import attr
+import datetime
+from dateutil.relativedelta import relativedelta
+from gevent.pool import Pool
+
+from django.core.cache import cache
+from django.core.management import CommandError
+from django.core.management.base import BaseCommand
+
+from custom.icds_reports.tasks import (
+    _aggregate_gm_forms, _aggregate_df_forms,_aggregate_cf_forms, _aggregate_ccs_cf_forms,
+    _aggregate_child_health_thr_forms, _aggregate_ccs_record_thr_forms, _aggregate_child_health_pnc_forms,
+    _aggregate_ccs_record_pnc_forms, _aggregate_delivery_forms, _aggregate_bp_forms,
+    _aggregate_awc_infra_forms, _child_health_monthly_table, _agg_ls_awc_mgt_form, _agg_ls_vhnd_form,
+    _agg_beneficiary_form, create_mbt_for_month, setup_aggregation, _agg_ls_table,
+    _update_months_table, _daily_attendance_table, _agg_child_health_table,
+    _ccs_record_monthly_table, _agg_ccs_record_table, _agg_awc_table, _agg_awc_table_weekly,
+    aggregate_awc_daily, email_dashboad_team
+)
+
+
+STATE_TASKS = {
+    'aggregate_gm_forms': _aggregate_gm_forms,
+    'aggregate_df_forms': _aggregate_df_forms,
+    'aggregate_cf_forms': _aggregate_cf_forms,
+    'aggregate_ccs_cf_forms': _aggregate_ccs_cf_forms,
+    'aggregate_child_health_thr_forms': _aggregate_child_health_thr_forms,
+    'aggregate_ccs_record_thr_forms': _aggregate_ccs_record_thr_forms,
+    'aggregate_child_health_pnc_forms': _aggregate_child_health_pnc_forms,
+    'aggregate_ccs_record_pnc_forms': _aggregate_ccs_record_pnc_forms,
+    'aggregate_delivery_forms': _aggregate_delivery_forms,
+    'aggregate_bp_forms': _aggregate_bp_forms,
+    'aggregate_awc_infra_forms': _aggregate_awc_infra_forms,
+    'agg_ls_awc_mgt_form': _agg_ls_awc_mgt_form,
+    'agg_ls_vhnd_form': _agg_ls_vhnd_form,
+    'agg_beneficiary_form': _agg_beneficiary_form,
+    'create_mbt_for_month': create_mbt_for_month
+}
+
+ALL_STATES = {
+    'child_health_monthly': _child_health_monthly_table
+}
+
+NORMAL_TASKS = {
+    'setup_aggregation': setup_aggregation,
+    'agg_ls_table': _agg_ls_table,
+    'update_months_table': _update_months_table,
+    'daily_attendance': _daily_attendance_table,
+    'agg_child_health': _agg_child_health_table,
+    'ccs_record_monthly': _ccs_record_monthly_table,
+    'agg_ccs_record': _agg_ccs_record_table,
+    'agg_awc_table': _agg_awc_table,
+    'aggregate_awc_daily': aggregate_awc_daily,
+}
+
+
+@attr.s
+class AggregationQuery(object):
+    by_state = attr.ib()
+    func = attr.ib()
+
+
+class Command(BaseCommand):
+    help = "Run portion of dashboard aggregation. Used by airflow"
+
+    def add_arguments(self, parser):
+        parser.add_argument('query_name')
+        parser.add_argument('date')
+        parser.add_argument('interval', type=int)
```

Comment:
It appears that this is attempting to use the old date & interval interface from the aggregation, and that date is being used as part of a cache key. I think that that's vulnerable to some conflicts if two aggregations are started at one time, and the date and interval math is a little confusing (partly because I would expect a date interval to be something like over 1 week, over 1 month, over 3 months and so on). 

What if this was changed to accept `aggregation_date` and `aggregation_uuid`? `aggregation_date` would serve as the current return value `get_agg_date` and would be more clear on what set of dates we are currently aggregating. `aggregation_uuid` would allow us to tag all actions taken by a single aggregation and would make it clearer when two are running in tandem. I'm assuming airflow will be able to generate a UUID for this and keep it throughout the the task.